[Stage] Remove focus from app when in spread mode

mariogrip opened the pull request:
If we enter spread, we should remove focus from any app

Focus should not be given back here as that is handled by spread when
selecting app.

This fixes: https://github.com/ubports/ubuntu-touch/issues/1236
Needs: https://github.com/ubports/unity8/pull/229

NeoTheThird commented:
![3i3rld](https://user-images.githubusercontent.com/12068939/69921046-ff569f80-148e-11ea-8ac4-80891c2f1713.jpg)

UniversalSuperBox commented:
If you class 49 tests as "Some," sure.

UniversalSuperBox commented:
Good thing those tests are there, too. There seem to be some problems with this PR as it stands now. I think they stem from:

```
file:///home/dalton/unity8/qml/Stage/Stage.qml:499:9: QML State: Binding loop detected for property "when"
```

mariogrip committed:
[TLWM] Change rootFocus to a property

This changes rootFocus to a qt property insted of a callable function

mariogrip committed:
[Stage] Remove focus from app when in spread mode

If we enter spread, we should remove focus from any app

Focus should not be given back here as that is handled by spread when
selecting app.

This fixes: ubports/ubuntu-touch#1236

UniversalSuperBox commented:
Closing in favor of https://github.com/ubports/unity8/pull/233
